rust/candid/src/parser/grammar.lalrpop

```diff
@@ -220,20 +225,20 @@ pub IDLProg: IDLProg = {
 
 Input: Input = {
     "text" => Input::Text(<>),
-    "blob" <"text"> => Input::Blob(<>),
+    "bytes" => Input::Blob(<>),
 }
 
-Assert: Assert = <id:"id"> <assert:Assertion> <desc:"text"?> =>? {
+Assert: Assert = <id:"id"> <assert:Assertion> =>? {
     if id != "assert" {
         Err(error("not an assert"))
-    } else { Ok(Assert { assert, desc }) }
+    } else { Ok(assert) }
 };
 
-Assertion: Assertion = {
-    <input:Input> ":" <typ:TupTyp> => Assertion::Decode(input, typ),
-    <input:Input> "!" ":" <typ:TupTyp> => Assertion::NotDecode(input, typ),
-    <left:Input> "=" "=" <right:Input> ":" <typ:TupTyp> => Assertion::Equal(left, right, typ),
-    <left:Input> "!" "=" <right:Input> ":" <typ:TupTyp> => Assertion::NotEqual(left, right, typ),
+Assertion: Assert = {
+    <left:Input> ":" <typ:TupTyp> <desc:"text"?> => Assert { left, right: None, typ, pass: true, desc },
+    <left:Input> "!" ":" <typ:TupTyp> <desc:"text"?> => Assert { left, right: None, typ, pass: false, desc },
```

Comment:
Would this allow
```
assert blob "foo" ! : ()
```
with a space inbetween `!:`?

Woudl `"!:"` work differently?

Reply:
Good catch. Better to have a dedicated token then.